### DISCUSSION TITLE
Error loading package - "illegal operand"

### DISCUSSION THREAD

**Discussion by kennaas** [ropensci/targets#993] [Help] [answered]
Hi! I am getting an "irrecoverable exception" in R when loading `targets`, and I am struggling to understand the error message (pasted below). Loading other R packages from the same install location works fine. I'm on `R` version 4.1.2 and the latest version of `targets`. Any advice?

> \>library(targets)
>
>  *** caught illegal operation ***
> address 0x7fbc761d5a73, cause 'illegal operand'
> 
> Traceback:
>  1: ps::ps_handle()
>  2: fun(libname, pkgname)
>  3: doTryCatch(return(expr), name, parentenv, handler)
>  4: tryCatchOne(expr, names, parentenv, handlers[[1L]])
>  5: tryCatchList(expr, classes, parentenv, handlers)
>  6: tryCatch(fun(libname, pkgname), error = identity)
>  7: runHook(".onLoad", env, package.lib, package)
>  8: loadNamespace(j <- i[[1L]], c(lib.loc, .libPaths()), versionCheck = vI[[j]])
>  9: asNamespace(ns)
> 10: namespaceImportFrom(ns, loadNamespace(j <- i[[1L]], c(lib.loc,     .libPaths()), versionCheck = vI[[j]]), i[[2L]], from = package)
> 11: loadNamespace(j <- i[[1L]], c(lib.loc, .libPaths()), versionCheck = vI[[j]])
> 12: asNamespace(ns)
> 13: namespaceImportFrom(ns, loadNamespace(j <- i[[1L]], c(lib.loc,     .libPaths()), versionCheck = vI[[j]]), i[[2L]], from = package)
> 14: loadNamespace(package, lib.loc)
> 15: doTryCatch(return(expr), name, parentenv, handler)
> 16: tryCatchOne(expr, names, parentenv, handlers[[1L]])
> 17: tryCatchList(expr, classes, parentenv, handlers)
> 18: tryCatch({    attr(package, "LibPath") <- which.lib.loc    ns <- loadNamespace(package, lib.loc)    env <- attachNamespace(ns, pos = pos, deps, exclude, include.only)}, error = function(e) {    P <- if (!is.null(cc <- conditionCall(e)))         paste(" in", deparse(cc)[1L])    else ""    msg <- gettextf("package or namespace load failed for %s%s:\n %s",         sQuote(package), P, conditionMessage(e))    if (logical.return && !quietly)         message(paste("Error:", msg), domain = NA)    else stop(msg, call. = FALSE, domain = NA)})
> 19: library(targets)

**Comment by wlandau** [comment, marked as answer]
Looks like the error is coming from `ps::ps_handle()`. I recommend you try calling `ps::ps_handle()` in a new R session and see if you get the same error. If `ps::ps_handle()` errors with the same "caught illegal operation" etc., please continue the discussion at https://github.com/r-lib/ps/issues. Reinstalling `ps` may solve the problem.

`targets` has no C, C++, or Fortran code, and it does not depend on the `ps` package. So a second thing to try would be to individually load each of the packages that `targets` depends on. If you get an error loading one of those packages, you can try reinstalling it, or you can move the discussion to the issue tracker of its development repository.

https://github.com/ropensci/targets/blob/d82ed1316f3715b29c4cad7f80ad10441f280524/DESCRIPTION#L56-L73
